frontend/src/container/PlannedDowntime/DropdownWithSubMenu/DropdownWithSubMenu.tsx

```diff
@@ -0,0 +1,229 @@
+import './DropdownWithSubMenu.styles.scss';
+
+import { CheckOutlined } from '@ant-design/icons';
+import { Button, Checkbox, Popover, Typography } from 'antd';
+import { CheckboxChangeEvent } from 'antd/es/checkbox';
+import { FormInstance } from 'antd/lib';
+import { useEffect, useState } from 'react';
+import { popupContainer } from 'utils/selectPopupContainer';
+
+import { recurrenceOptions } from '../PlannedDowntimeutils';
+
+interface SubOption {
+	label: string;
+	value: string;
+}
+
+export interface Option {
+	label: string;
+	value: string;
+	submenu?: SubOption[];
+}
+
+interface DropdownProps {
+	options: Option[];
+	form: FormInstance<any>;
+	setRecurrenceOption: React.Dispatch<
+		React.SetStateAction<string | undefined | null>
+	>;
+}
+
+export function DropdownWithSubMenu(props: DropdownProps): JSX.Element {
+	const { options, form, setRecurrenceOption } = props;
+	const [selectedOption, setSelectedOption] = useState<Option | null>(
+		!form.getFieldValue('recurrenceSelect')
```

Comment:
negate the condition for better readability

Reply:
done